apis/v1alpha1/httproute_types.go

```diff
@@ -115,9 +115,8 @@ const (
 	HeaderMatchTypeImplementionSpecific = "ImplementationSpecific"
 )
 
-// HTTPRouteMatch defines the predicate used to match requests to a
-// given action.
-type HTTPRouteMatch struct {
+type HTTPPathMatch struct {
```

Comment:
Although I understand the desire to have path match be consistent with header match, I think this ends up making a more significant change than we want. This change results in path match becoming optional where before it was defaulted to match `/`. It also results in an ambiguous situation where nothing is actually required or set by default in `HTTPRouteMatch`. To avoid that, I'd lean towards only changing header matching with this PR.

Reply:
Yep, we want to be careful about the semantics here. I'm OK with dropping the '/' path match because implicitly requiring an exact match on '/' seems odd now that other match types are available. Being able to produce an empty HTTPRouteMatch is weird; you could argue equally that it should match everything or nothing.

Reply:
Yeah I completely agree here, but had a slightly different idea for a solution. I filed a [follow up issue](https://github.com/kubernetes-sigs/service-apis/issues/255) that suggests we should be defaulting to a `Prefix` match here so the default `/` would match all paths.

Reply:
+1 on Rob's proposal of leaving path match in the `HTTPRouteMatch` struct and defaulting to `'type: Prefix, path: "/"'`.  It would make the spec with default match simpler and less ambiguous.

Reply:
I've been wondering for a few weeks and I can't recall the reason of why we put in `Exact` match as the default for `PathMatchType` because it seems draconian. Anyone recall the reason?

I agree with Rob and Yi here. Header based routing in addition to path based is more common in practice and less error-prone.

Reply:
> Although I understand the desire to have path match be consistent with header match, I think this ends up making a more significant change than we want. 

I mulled this over and I think that we should make these consistent. Maning the API consistent and predictable is really important, and having different syntaxes for different kind of match actually adds complexity rather than removing it.


This is what you have to do now, which is inconsistent, a bit ugly (`patchMatchType`), and really needs handy reference to the docs to write:
```yaml
   - match:
         path: /bar
         pathMatchType: Prefix
         headers:
           type: Contains
           values:
              foo: bar
```

Reply:
Resolving as per WG discussion and consensus in favor of this.